Control/Monad/Error/Class.hs

```diff
@@ -190,3 +194,21 @@ instance (Monoid w, MonadError e m) => MonadError e (LazyWriter.WriterT w m) whe
 instance (Monoid w, MonadError e m) => MonadError e (StrictWriter.WriterT w m) where
     throwError = lift . throwError
     catchError = StrictWriter.liftCatch catchError
+
+-- | MonadError analog to the 'try' function.
+tryError :: MonadError e m => m a -> m (Either e a)
+tryError action = (liftM Right action) `catchError` (return . Left)
+
+-- | Modify the value (but not the type) of an error.  The type is
+-- fixed because of the functional dependency @m -> e@.  If you need
+-- to change the type of @e@ use 'mapError'.
+withError :: MonadError e m => (e -> e) -> m a -> m a
+withError f action = tryError action >>= either (throwError . f) return
+
+-- | As 'handle' is flipped 'catch', 'handleError' is flipped 'catchError'.
+handleError :: MonadError e m => (e -> m a) -> m a -> m a
+handleError = flip catchError
+
+-- | MonadError analogue of the 'mapExceptT' function.
```

Comment:
It might be wise to recap what this function does (e.g., "Map the unwrapped computation using the given function").

Also, s/`MonadError`/`'MonadError'`/

Reply:
resolved